Content.Server/Radar/RadarConsoleSystem.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using Content.Server.Solar.Components;
+using Content.Server.UserInterface;
+using Content.Shared.Radar;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+
+namespace Content.Server.Radar;
+
+public class RadarConsoleSystem : EntitySystem
+{
+    [Dependency] private readonly IMapManager _mapManager = default!;
+
+    private static float Frequency = 1.5f;
+
+    private float _accumulator;
+
+    public override void Update(float frameTime)
+    {
+        _accumulator += frameTime;
+
+        if (_accumulator < Frequency)
+            return;
+
+        _accumulator -= Frequency;
+
+        foreach (var component in EntityManager.EntityQuery<RadarConsoleComponent>())
+        {
+            var s = component.Owner.GetUIOrNull(RadarConsoleUiKey.Key);
+            var radarTransform = Transform(component.Owner);
+            var radarPos = radarTransform.WorldPosition;
+            var radarRot = radarTransform.WorldRotation;
+
+            if (s is null)
+                continue;
+
+            var map = Transform(component.Owner).MapID;
+
+            var objects = new List<RadarObjectData>();
+
+            foreach (var grid in _mapManager.GetAllMapGrids(map))
+            {
+                var phy = Comp<PhysicsComponent>(grid.GridEntityId);
+                var transform = Transform(grid.GridEntityId);
```

Comment:
Just use FindGridsIntersecting and specify a range because this will also have the problem Moony's fork has of dying with high grid counts otherwise (whenever the check isn't O(n)) and I imagine this is a performance problem on her fork.

Realistically if you wanted based levels of performance you'd make a collider for the radar and subscribe to collision events which way down the line we'll need to do for high ship counts.

Reply:
Okay GridTrees is also a thing now so definitely just slap FindGridsIntersecting on it for now.

Reply:
Yea the original reason I did iteration was because there was 300 grids and FindGridsIntersecting didn't seem like it'd be faster than just iterating all of them.